### DISCUSSION TITLE
Stacked buffer

### DISCUSSION THREAD

**Discussion by noahfraiture** [helix-editor/helix#11961] [Ideas]
## Hello everyone,

I would like to propose an exciting feature for helix: the addition of stacked buffers, similar to the functionality found in Zellij or Aerospace on mac (never tested). This feature would allow users to access multiple buffers seamlessly through existing panel motion commands.

### Key Features:
- Single Visible Buffer: Users would see only one or two buffer at a time, but could easily switch to hidden buffers by moving to adjacent windows. Because with small screen, more than two buffer is often not usable and helix can't use multiple screen.
- Visual Indicators: We could implement visual cues to signal when a hidden buffer is accessible, enhancing the user experience.
- Multiple Instances of the Same File: This approach allows for multiple instances of the same file without cluttering the workspace with all visible buffers.
- Simplified Navigation: A structured layout would make navigation much easier, reducing reliance on file or buffer pickers.
- Synergy with Plugins: This design would also create a strong synergy with potential plugins, such as terminal integrations, that could operate like windows.

I look forward to hearing your thoughts!

**Comment by TornaxO7** [comment]
I could imagine to have this as a plugin rather than as a core feature, but that are just my two cents.

**Comment by seifertm** [comment]
My understanding is that you'd like to see a functionality like Zellij's stacked panes or a window manager to Helix. In order to give the maintainers a basis for accepting your idea, maybe it helps to describe what problem you want to solve.

@noahfraiture Can you describe a use case that's possible with your proposed functionality that is not achievable with a combination of Helix and Zellij right now?

**Comment by noahfraiture** [comment]
Yes of course @seifertm .
Situation : I have a large code base and a small scree, I frequently jump between different section of the code (let's say 3), in the same file or in different file. Since my screen is not that large, and helix is limited to a single screen, I can't keep all the 3 buffer I want, and I have to close one because my screen is too small to display more than two buffer and keep them readable. 
There's two solution currently:
- Buffer picker. But it's more like an history of buffer, you can't have multiple for the same file.
- Zellij plane. It require yet another tool, and it need you to start a new helix instance. This is not very good because the clipboard, undo history and every helix feature, won't be shared accross sessions

The third solution would be to have stacked buffer. You keep them open, but not display as their would be "hide behind others" or "hide on the side of you screen". 
Another advantage is the support of plugin it will give. Filetree and terminal would beneficiate a lot from this feature, as you could keep your filetree on a stacked pane on the left that you cuold easily reach with window motion.

**Reply by seifertm** [reply]
Thanks, that made it more clear, for me at least!

I think #312 could be related, because it would enable your second solution to communicate across different Helix processes in different windows.

**Reply by noahfraiture** [reply]
Yes this is another possible solution, it would at least solve the issue that the zellij solution have.However it doesn't seem to move a lot
